### PR TITLE
fix: bump flutter version

### LINKED ISSUE #97: support the dart 3 and flutter 3.10

### PR DESCRIPTION
**Pull request by Harshit933**
This commit bumps the flutter version to support till the flutter 4.0.0

fixes #97

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ publish_to: 'none' # Remove this line if you wish to publish to pub.dev
 version: 0.0.1-alpha.3+1
 
 environment:
-  sdk: ">=2.12.3 <3.0.0"
+  sdk: ">=3.0.0 <4.0.0"
```
Comment (vincenzopalazzo) [inline pubspec.yaml:8]:
```suggestion
  sdk: ">=3.0.0 <4.0.0"
```

we not bump till >2.15? will some requirements stop us from using dart 2.x?

Comment (Harshit933) [reply pubspec.yaml:8]: I tried running the app after bumping to flutter 3. It works fine but we can downgrade the sdk to support larger dependency.

Comment (vincenzopalazzo) [reply pubspec.yaml:8]: I think `">=2.12.3 <4.0.0"` woul be nice

I accepted a similar PR yesterday in https://github.com/zino-hofmann/graphql-flutter/pull/1328

Comment (Harshit933) [reply pubspec.yaml:8]: Sure!

**Review by vincenzopalazzo — COMMENTED** [review]
Can you also fix the make file?

The CI is complaining about this

```
Run flutter format --set-exit-if-changed .
The "format" command is deprecated. Please use the "dart format" sub-command instead, which has the same command-line usage as "flutter format".

```

The solution is at this line https://github.com/dart-lightning/lndart.clnapp/blob/main/Makefile#L10 and changing the `dart $(FMT) .` to `$(CC) $(FMT) .`

**Comment by Harshit933** [comment]
Changing the command and then running `make` gives the following error.

```flutter format .
The "format" command is deprecated. Please use the "dart format" sub-command instead, which has the same command-line usage as "flutter format"```

**Comment by vincenzopalazzo** [comment]
@Harshit933 ok so we should put the `subosito/flutter-action@v2` in https://github.com/dart-lightning/lndart.clnapp/blob/main/.github/workflows/build-ci.yml#LL18C15-L18C41

We was running an old flutter, take a look to the other yml file as well

**Comment by vincenzopalazzo** [comment]
This commit https://github.com/dart-lightning/lndart.clnapp/pull/100/commits/f08134c1097ee932c65474f94071e49489b70e1e does not need to stay in the changelog

you can write `ci: upgrade flutter action`

**Comment by Harshit933** [comment]
> This commit [f08134c](https://github.com/dart-lightning/lndart.clnapp/commit/f08134c1097ee932c65474f94071e49489b70e1e) does not need to stay in the changelog
> 
> you can write `ci: upgrade flutter action`

What should I do here? Make a new PR or rename the previous commit. I am a bit confused.

**Review by vincenzopalazzo — APPROVED** [review]

**Pull request merged**